### PR TITLE
Introduce a global router, and add basic routing logic

### PR DESCRIPTION
**Pull request by Others**
Next step is creating data structures, and adding actual responses that make sense

### CODE REVIEW

File: src/request_handler.rs [hunk 1]
```diff
@@ -1,7 +1,45 @@
...
+    fn new() -> HttpRequestHandler {
+        HttpRequestHandler {}
+    }
+
+    fn handle(&self, req: Request<Body>) -> Response<Body> {
+        let method = req.method();
+        let uri = req.uri();
+        let query = uri.query().unwrap_or("");
+
+        // Get the uri path, and then split it around slashes into components
```
Comment (pcodes) [inline src/request_handler.rs]: What are you skipping/why?

File: src/request_handler.rs [hunk 2]
```diff
@@ -1,7 +1,45 @@
...
+        let path_components: Vec<&str> = uri.path().split('/').skip(1).collect();
+
+        let response_body = if path_components[0] == "meta" {
```
Comment (pcodes) [inline src/request_handler.rs]: Might want to include a comment with a sample uri

File: src/request_handler.rs (hunk 1 above)
Comment (Others) [reply src/request_handler.rs]: will add comment in next PR

File: src/request_handler.rs (hunk 2 above)
Comment (Others) [reply src/request_handler.rs]: Will do!

**Pull request merged**